akka-actor-tests/src/test/java/akka/actor/ActorCreationTest.java

```diff
@@ -287,5 +315,18 @@ public void testPropsUsingCreatorWithoutClass() {
     assertEquals(UntypedTestActor.class, p.actorClass());
   }
 
+  @Test
+  public void testIssue20537Reproducer() {
+    final Issue20537Reproducer.ReproducerCreator creator = new Issue20537Reproducer.ReproducerCreator(false);
+    final Props p = Props.create(creator);
+    assertEquals(Issue20537Reproducer.class, p.actorClass());
+
+    ArrayList<Props> pList = IntStream.range(0, 4).mapToObj(i -> Props.create(creator))
+        .collect(toCollection(ArrayList::new));
```

Comment:
the lambda doesn't matter for reproducing the issue, but I included that test anyway